app/models/deploy_service.rb

```diff
@@ -78,6 +78,7 @@ def latest_approved_deploy(reference, project)
 
   def copy_approval_from_last_deploy(deploy)
     return false unless last_deploy = latest_approved_deploy(deploy.reference, deploy.stage.project)
+    return false if deploy.changeset_to(last_deploy).commits.any?
```

Comment:
might be cheaper to resolve commit and check if it is not the same ... also make sure changeset_to does not compare refs which would be the same

Reply:
We may cache the changeset in the previous Review step, it should re-use the cache.